### ISSUE TITLE
Variance prediction in regression kriging

### ISSUE THREAD

**Issue by alexisrosuel** [GeoStat-Framework/PyKrige#66]
That would be useful to return not only the point estimate in regression kriging, but also the variance ? So that the user can then compute a confidence interval of each prediction...

I think it would not be too difficult : 
- regression model based on scikit doesn't return this. So let's not consider it for the moment
- the kriging model **does** return it ! So we could just use it directly in rk

It's not perfect, since we will not catch all the variance, but it's better than nothing...

What do you think ?

**Comment by rth** [comment]
> regression model based on scikit doesn't return this

Well [Gaussian process regression](http://scikit-learn.org/stable/modules/generated/sklearn.gaussian_process.GaussianProcessRegressor.html#sklearn.gaussian_process.GaussianProcessRegressor) from scikit-learn does return a variance, but I guess you are right, one could use a krigging model in rk, I imagine. Generally this is more a question for @basaks ..

Off topic, it would be great to have a clear explanation somewhere in the docs on what are the differences between the kriging algorithms found in PyKrige and Gaussian Regression Regression algorithms found in machine learning libraries (e.g. scikit-learn) (see e.g. related stack overflow questions  [[1](https://stats.stackexchange.com/questions/48405/confusion-related-to-difference-of-kriging-and-gaussian-processes)], [[2](https://stats.stackexchange.com/questions/30644/queries-regarding-gaussian-processes-and-kriging)])

**Comment by bsmurphy** [comment]
@basaks should comment on the actual content of this thread... but to comment on @rth 's second point, agreed that would be useful... I can work on that as well

**Comment by basaks** [comment]
I think of the gaussian process in sklearn is like simple kriging, where you know the mean.
[The following line from the sklearn docs support indicates that too.](http://scikit-learn.org/stable/modules/gaussian_process.html)

> The prior mean is assumed to be constant and zero (for normalize_y=False) or the training data’s mean (for normalize_y=True). 


Finally in relation to the regression kriging implementation in PyKrige:

> If the residuals from MLR exhibit spatial dependence, an estimate of the value of the regression residuals at the unknown locations can be made using OK. This process was referred to as the RK by Odeh and McBratney (1994); Odeh et al., 1994).

So our implementation of regression kriging in PyKrige is regression kriging (RK) according to this definition, if we used MLR (like elasticnet) as the regression step followed by OK to krige the residuals.

**Comment by rth** [comment]
> I think of the gaussian process in sklearn is like ...

@basaks Looks like a documentation PR waiting to happen :) Maybe we could do a  `doc/faq.md` (or something similar) ? Some tips on how to choose the variogram model (and between the kriging models)  would be quite useful too IMO ..

**Comment by MuellerSeb** [comment]
I think, we could refer to the GSTools documentation, where some further information can be provided.

Closing for now. Feel free to re-open.
